app/notebooks/question_answer_generator/generate_q_a_pairs.py

```diff
@@ -0,0 +1,95 @@
+import csv
+import json
+from litellm import completion
+
+import os
+from pydantic import BaseModel
+
+from src.app_config import app_config
+from src.db.models.document import Chunk, Document
+from src.generate import completion_args
+
+
+GENERATE_QUESTION_ANSWER_PROMPT = """
+Using the provided text, generate at least 5 unique questions and answers, avoid rephrasing or changing the punctuation of the question to ensure distinct questions and answers.
+Respond with a list of JSON dictionaries in the following format  (do not wrap in JSON markers):
+question: The generated question based on the content.
+answer: The answer to the question, derived from the content.
+document_name: The name of the document.
+document_source: The source or URL of the document.
+document_id: the id of the chunk or document.
+Example
+question: What's a base period for SDI?
```

Comment:
```suggestion
Respond with a list of JSON dictionaries in the following format (do not wrap in JSON markers):
question: The generated question based on the content.
answer: The answer to the question, derived from the content.
document_name: The name of the document.
document_source: The source or URL of the document.
document_id: the id of the chunk or document.

Example
question: What's a base period for SDI?
```

Since the response is structured, why not have it to produce JSON?

Reply:
I've revised the prompt several times with the term `Your response should return valid JSON` and provided some examples but the response always comes back as a string.